### PR TITLE
Add NSIMD library

### PR DESCRIPTION
**Pull request by JDPailleux**
Hello,
We have added the NSIMD installation in compiler_explorer.
We did a Pull Request on 'infra'. 
Thank you in advance.

### CODE REVIEW

**Comment by mattgodbolt** [comment]
Waiting on a reply on https://github.com/compiler-explorer/infra/pull/412 to get that merged.

File: etc/config/cuda.amazon.properties [hunk 1]
```diff
@@ -56,3 +56,11 @@ libs.cucub.versions=180
...
 libs.cucub.versions.180.version=1.8.0
 libs.cucub.versions.180.path=/opt/compiler-explorer/libs/cub/1.8.0
+
+libs=nsimd
+libs.nsimd.name=NSIMD
+libs.nsimd.versions=cuda
+libs.nsimd.url=https://github.com/agenium-scale/nsimd/
+libs.nsimd.versions.cuda.version=cuda
+libs.nsimd.versions.cuda.path= /opt/compiler-explorer/libs/nsimd/spmd/cuda/include
+libs.nsimd.versions.cuda=/opt/compiler-explorer/libs/nsimd/spmd/cuda/lib
```
Comment (mattgodbolt) [inline etc/config/cuda.amazon.properties:66]: Super nitpick: can you ensure there's a newline at the end of the file please?

**Review by mattgodbolt — APPROVED** [review]
Looks good once we're all set on the `infra` merge. Thank you very much!

File: etc/config/cuda.amazon.properties (hunk 1 above)
Comment (JDPailleux) [reply etc/config/cuda.amazon.properties:66]: Of course, I will add the new line right away.

**Comment by mattgodbolt** [comment]
Oh golly, I can't believe I let this one slip. I'm so sorry!

**Comment by mattgodbolt** [comment]
D'oh of course; this is the library we're still waiting on in infra :) I had a moment of panic thinking "why on earth didn't we merge this already!?" :) Will mark as waiting until we get the infra merge.

**Comment by apmorton** [comment]
The infra side of this is all set, and nsimd v2.2 is installed on prod; I don't have time to test this side of the PR now, but I will circle back to this later if someone else doesn't get to it before me.

File: etc/config/c++.amazon.properties [hunk 2]
```diff
@@ -1446,6 +1446,21 @@ libs.immer.versions=trunk
 libs.immer.versions.trunk.version=trunk
 libs.immer.versions.trunk.path=/opt/compiler-explorer/libs/immer/trunk
 
+libs=nsimd
+libs.nsimd.name=NSIMD
+libs.nsimd.url=https://github.com/agenium-scale/nsimd/
+libs.nsimd.versions=x86_64:arm:arm64
```
Comment (apmorton) [inline etc/config/c++.amazon.properties:1452]: These versions don't feel correct to me - what do we expect to happen once the next version of nsimd is released?
Typically we keep around all tagged versions of libraries and add newer versions.

I think we have to do something like `libs.nsimd.versions.x86_64-22`

Comment (gquintin) [reply etc/config/c++.amazon.properties:1452]: @apmorton ok to do that, do we need to do something for you concerning this?

**Comment by partouf** [comment]
merged with #2433

**Comment by partouf** [comment]
This is now live!

**Pull request closed without merging**